File: app/src/main/java/com/apache/fastandroid/setting/SettingActivity.java
package com.apache.fastandroid.setting;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.preference.PreferenceFragment;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.apache.fastandroid.R;

/**
 * Created by jerryliu on 2017/6/24.
 */

public class SettingActivity extends AppCompatActivity {
    public static void launch(Activity from){
        from.startActivity(new Intent(from,SettingActivity.class));
    }
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //getActionBar().setTitle("设置");
        SettingFragment settingFragment = new SettingFragment();
        getFragmentManager().beginTransaction()
                .add(android.R.id.content,settingFragment)
                .commit();
    }

    public static class SettingFragment extends PreferenceFragment {
        @Override
        public void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            // 加载xml资源文件
            addPreferencesFromResource(R.xml.fragment_setting);
        }
    }

}

File: app/src/main/java/com/apache/fastandroid/setting/SettingActivity.kt
package com.apache.fastandroid.setting

import android.os.Bundle
import androidx.preference.ListPreference
import androidx.preference.PreferenceFragmentCompat
import com.apache.fastandroid.R
import com.apache.fastandroid.databinding.ActivitySettingsBinding
import com.apache.fastandroid.util.GlobalConstans
import com.apache.fastandroid.util.GlobalValues
import com.tesla.framework.ui.activity.BaseVBActivity

/**
 * Created by Jerry on 2022/5/2.
 */
class SettingActivity : BaseVBActivity<ActivitySettingsBinding>(ActivitySettingsBinding::inflate) {

    override fun layoutInit(savedInstanceState: Bundle?) {
        super.layoutInit(savedInstanceState)

    }

    class SettingFragment:PreferenceFragmentCompat(){
        init {

        }
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.settings,rootKey)

            findPreference<ListPreference>(GlobalConstans.SpKey.WORK_MODE)?.apply {
                setOnPreferenceChangeListener { preference, newValue ->
                    GlobalValues.workMode = newValue as String
                    true
                }
            }
        }


    }
}